### PR TITLE
Update KeyVault to T2 package.

### PR DESCRIPTION
**Pull request by FrankieTF**
Updated target KeyVaultManager:
- KeyVaultClient updated to SecretClient and CertificateClient. 
- Updated Microsoft.Extensions.Configuration.AzureKeyVault to Azure.Extensions.AspNetCore.Configuration.Secrets

### CODE REVIEW

File: src/GraphWebhooks-Core/Models/KeyVaultOptions.cs [hunk 1]
```diff
@@ -9,5 +9,6 @@ public class KeyVaultOptions
         public string ClientId { get; set; }
         public string ClientSecret { get; set; }
         public string CertificateUrl { get; set; }
+        public string KeyVaultUri { get; set; }
```
Comment (baywet) [inline src/GraphWebhooks-Core/Models/KeyVaultOptions.cs:12]: Do we need both the KeyVault URL and the certificate URL?
Couldn't we:
- deduce the keyvault URL from the certificate URL -OR-
- ask for the certificate path 

?

**Review by baywet — CHANGES_REQUESTED** [review]

File: src/GraphWebhooks-Core/Models/KeyVaultOptions.cs (hunk 1 above)
Comment (FrankieTF) [reply src/GraphWebhooks-Core/Models/KeyVaultOptions.cs]: Deducing the keyvault URL from the certificate URL

File: src/GraphWebhooks-Core/KeyVaultManager.cs [hunk 2]
```diff
@@ -59,7 +59,9 @@ private async Task GetCertificateFromKeyVault()
                 string clientId = KeyVaultOptions.Value.ClientId;
                 string clientSecret = KeyVaultOptions.Value.ClientSecret;
                 string certificateUrl = KeyVaultOptions.Value.CertificateUrl;
-                string keyVaultUri = KeyVaultOptions.Value.KeyVaultUri;
+                
+                string[] splitCertificateUrl = certificateUrl.Split("/");
```
Comment (baywet) [inline src/GraphWebhooks-Core/KeyVaultManager.cs:63]:
```suggestion
                string[] splitCertificateUrl = certificateUrl.Split("/", StringSplitOptions.RemoveEmptyEntries);
```

File: src/GraphWebhooks-Core/KeyVaultManager.cs [hunk 3]
```diff
@@ -59,7 +59,9 @@ private async Task GetCertificateFromKeyVault()
...
+                string keyVaultUri = splitCertificateUrl[0] + "//" + splitCertificateUrl[2];
```
Comment (baywet) [inline src/GraphWebhooks-Core/KeyVaultManager.cs:64]:
```suggestion
                string keyVaultUri = splitCertificateUrl[0] + "//" + splitCertificateUrl[1];
```

File: .github/workflows/codeql-analysis.yml [hunk 4]
```diff
@@ -46,9 +44,9 @@ jobs:
     #    and modify them (or add more) to build your code if your project
     #    uses a compiled language
 
-    #- run: |
-    #   make bootstrap
-    #   make release
+    - run: |
+       make bootstrap
```
Comment (baywet) [inline .github/workflows/codeql-analysis.yml:48]: this is a dotnet repo, we should probably call dotnet build at this stage instead. And maybe add a dotnet install/init step before the codeql initialization like here https://github.com/microsoftgraph/aspnetcore-webhooks-sample/blob/b04b3289f9cc24182da72406a2795eb1b1daeeee/.github/workflows/dotnetcore.yml#L16

**Review by baywet — CHANGES_REQUESTED** [review]

File: README.md [hunk 5]
```diff
@@ -213,6 +213,26 @@ Keep the console open while testing. If you close it, the tunnel also closes and
       - For the **ClientSecret** key, replace *ENTER_YOUR_SECRET* with the key of your registered Azure application. Note that in production apps you should always use certificates as your application secrets, but for this sample we will use a simple shared secret password.
       - For the **Tenantd** key, replace *ENTER_YOUR_TENANT_ID* with your tenant ID, this information can be found on the application registration page.  
 
+1. Set the environment variable for DefaultAzureCredential.
+
+   Follow one of the examples below depending on your operating system to create the environment variable. If using Windows close your open IDE or shell and restart it to be able to read the environment variable.
```
Comment (baywet) [inline README.md:218]: could we use dotnet secrets instead and provide a single, cross-platform, set of commands?
https://www.twilio.com/blog/2018/03/protecting-your-passwords-with-the-net-secret-manager.html

Comment (FrankieTF) [reply README.md]: if we use dotnet secrets instead, we could specified the credential. These credentials dont need commands to set environment variables.

File: .github/workflows/codeql-analysis.yml (hunk 4 above)
Comment (FrankieTF) [reply .github/workflows/codeql-analysis.yml]: Thanks, it helps a lot.

File: src/GraphWebhooks-Core/appsettings.json [hunk 6]
```diff
@@ -9,7 +9,7 @@
   "KeyVaultSettings": {
     "ClientId": "",
     "ClientSecret": "",
-    "CertificateUrl": ""
+    "CertificateUrl": "https://franktestkv.vault.azure.net/certificates/ChangeNotificationSampleCertificate"
```
Comment (baywet) [inline src/GraphWebhooks-Core/appsettings.json:12]:
```suggestion
    "CertificateUrl": ""
```

File: .github/workflows/codeql-analysis.yml [hunk 7]
```diff
@@ -26,6 +26,8 @@ jobs:
     # the head of the pull request instead of the merge commit.
     - run: git checkout HEAD^2
       if: ${{ github.event_name == 'pull_request' }}
+    
+    - name: Setup .NET Core
```
Comment (baywet) [inline .github/workflows/codeql-analysis.yml:30]:
```suggestion
    - name: Setup .NET Core
      uses: actions/setup-dotnet@v1
      with:
        dotnet-version: 3.1.101
```

**Review by baywet — CHANGES_REQUESTED** [review]

**Review by baywet — APPROVED** [review]

**Pull request merged**